src/main/java/com/sourcegraph/webhook/README.md

```diff
@@ -0,0 +1,66 @@
+# Webhooks
+The Sourcegraph Bitbucket plugin provides an internal implementation for webhooks that supports listening to events at a global or project scope.
+
+## Contents
+- [Payload](#payload)
+- [REST API](#rest-api)
+    * [List](#list)
+    * [Create](#create)
+    * [Delete](#delete)
+
+## Payload
+The event payload request is sent when an event is fired. There will be five attempts before the request fails. This will be recorded in the logs.
+
+The request has these headers:
+- `X-Event-Key` - contains the specific event key
+- `X-Hook-ID` - webhook id
+- `X-Hook-Name` - webhook name
+- `X-Hub-Signature` - HMAC SHA256 hash of the payload (request body)
+
+The payload is stored in the request body as a JSON object. They follow the schema specified in [Event Payload](https://confluence.atlassian.com/bitbucketserver0516/event-payload-966061436.html?utm_campaign=in-app-help&utm_medium=in-app-help&utm_source=stash#Eventpayload-repositoryevents) and
+[Bitbucket Entities](https://docs.atlassian.com/bitbucket-server/docs/5.16.0/reference/javascript/JSON.html) for each event.
+
+Example (`pr:merged`):
+<img width="1266" alt="Screen Shot 2019-11-13 at 6 21 57 AM" src="https://user-images.githubusercontent.com/3507526/68772049-fc7a3480-05dd-11ea-9676-707b40fd3daf.png">
```

Comment:
What do you think about pasting the actual JSON here? Image is very hard to copy.